### DISCUSSION TITLE
Argon2 parallelism and non-16 salt lengths

### DISCUSSION THREAD

**Discussion by Tronic** [jedisct1/libsodium#1115] [Q&A]
It seems that libsodium lacks some of the options that Argon2 hash and the reference implementation of it offer, making it impossible to write compatible code for passwords hashed via other libraries using parallelism > 1 or any other salt length than 16 bytes.

I would suggest adding these options, unless they are already supported and I have overlooked something. Similarly, it may be problematic that libsodium takes the memory requirement in bytes, while other implementations use kilobytes.

Related to #717.

**Comment by jedisct1** [comment]
`crypto_pwhash_str_verify()` can verify passwords hashed with any parallelism value, and any salt length.
